lib/profile/edit_profile/sections/contact/edit_contact.dart

```diff
@@ -1,91 +1,86 @@
-      child: Column(
-        children: <Widget>[
-          InputField(
-            labelText: editContactFullname, 
-            formField: buildFullNameField()
-          ),
-          InputField(
-            labelText: editContactEmailAddress, 
-            formField: buildEmailAddressField()
-          ),
-          InputField(
-            labelText: editContactPhoneNumber, 
-            formField: buildPhoneNumberField()
-          ),
-          InputField(
-            labelText: editContactWebsite, 
-            formField: buildWebsiteField()
-          ),
-        ],
+    store = StoreProvider.of<AppState>(context);
+
+    return StoreConnector<AppState, Contact>(
+      converter: contactSelector,
+      builder: (context, contact) => Container(
+        padding: EdgeInsets.only(bottom: 20.0),
+        child: Column(
+          children: <Widget>[
+            InputField(
+              labelText: editContactFullname, 
+              formField: buildFullNameField()
+            ),
+            InputField(
+              enabled: false,
+              labelText: editContactEmailAddress, 
+              formField: buildEmailAddressField(contact)
+            ),
+            InputField(
+              labelText: editContactPhoneNumber, 
+              formField: buildPhoneNumberField(contact)
+            ),
+            InputField(
+              labelText: editContactWebsite, 
+              formField: buildWebsiteField(contact)
+            ),
+          ],
+        )
       )
     );
   }
 
   buildFullNameField() {
-    return FormBuilderTextField(
-      attribute: 'fullName',
-      initialValue: fullName,
-      decoration: fieldDecoration(),
-      style: fieldTextStyle,
-      onChanged: (value) => setState(() { fullName = value as String; }),
-      validators: [
-        FormBuilderValidators.required(),
-        FormBuilderValidators.maxLength(100),
-      ],
-      keyboardType: TextInputType.text,
+    return StoreConnector<AppState, User>(
+      converter: userSelector,
+      builder: (context, user) => FormBuilderTextField(
+        attribute: 'fullName',
+        initialValue: user.fullName,
+        decoration: fieldDecoration(),
+        style: fieldTextStyle,
+        onChanged: (value) => store.dispatch(UpdateFullName(value)),
+        validators: [
+          FormBuilderValidators.required(),
+          FormBuilderValidators.maxLength(100),
+        ],
+        keyboardType: TextInputType.text,
+      )
     );
   }
 
-  buildEmailAddressField() {
+  buildEmailAddressField(Contact contact) {
     return FormBuilderTextField(
+      readOnly: true,
       attribute: 'emailAddress',
-      initialValue: contact['emailAddress'],
+      initialValue: contact.emailAddress,
       decoration: fieldDecoration(),
-      style: fieldTextStyle,
-      onChanged: (value) => setState(() { 
-        contact['emailAddress'] = value as String; 
-      }),
+      style: labelTextStyle(isEnabled: false),
```

Comment:
I realized that the user shouldn't be able to edit their emailAddress since it is a sole part of the verification process...? I disabled the field in the EditProfile page for now and we can talk more about it during one of our standups

Reply:
Yeah, exactly!! 💯